### PR TITLE
bsdlib: release 0.5.0

### PR DESCRIPTION
**Pull request by lemrey**
<h2>bsdlib release 0.5.0</h2>

* `bsd_irrecoverable_error_handler()` has been removed.
  The application now longer needs to implement it to receive errors
  during initialization, which are instead reported via `bsd_init()
  return value.
* `bsd_shutdown()` now returns an integer.
* Added RAW socket support.
* Added missing AGPS data models.
* Added APGS nofification support.
* Fixed an issue where AGPS data can not be written
  when the GPS socket is in stopped state
* Fixed a memory leak in the GPS socket

Signed-off-by: Emanuele Di Santo <emdi@nordicsemi.no>

### CODE REVIEW

File: bsdlib/CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,19 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+bsdlib 0.5.0
+************
+
+* `bsd_irrecoverable_handler()` has been removed.
+  The application now longer needs to implement it to receive errors during initialization, which are instead reported via `bsd_init() return value.
```
Comment (b-gent) [inline bsdlib/CHANGELOG.rst]:
```suggestion
  The application no longer needs to implement it to receive errors during initialization, which are instead reported via :cpp:func:`bsd_init()` return value.
```

File: bsdlib/CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,19 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+bsdlib 0.5.0
+************
+
+* `bsd_irrecoverable_handler()` has been removed.
```
Comment (b-gent) [inline bsdlib/CHANGELOG.rst]:
```suggestion
* :cpp:func:`bsd_irrecoverable_handler()` has been removed.
```

File: bsdlib/CHANGELOG.rst [hunk 3]
```diff
@@ -5,6 +5,19 @@ Changelog
...
+  The application now longer needs to implement it to receive errors during initialization, which are instead reported via `bsd_init() return value.
+* `bsd_shutdown()` now returns an integer.
```
Comment (b-gent) [inline bsdlib/CHANGELOG.rst]:
```suggestion
* :cpp:func:`bsd_shutdown()` now returns an integer.
```

File: bsdlib/CHANGELOG.rst [hunk 4]
```diff
@@ -5,6 +5,19 @@ Changelog
...
+* Added RAW socket support.
+* Added missing AGPS data models.
+* Added APGS nofification support.
```
Comment (b-gent) [inline bsdlib/CHANGELOG.rst]:
```suggestion
* Added APGS notification support.
```

File: bsdlib/CHANGELOG.rst [hunk 5]
```diff
@@ -5,6 +5,19 @@ Changelog
...
+* Fixed an issue where AGPS data can not be written when the GPS socket is in stopped state.
```
Comment (b-gent) [inline bsdlib/CHANGELOG.rst]:
```suggestion
* Fixed an issue where AGPS data could not be written when the GPS socket was in stopped state.
```

File: bsdlib/include/nrf_socket.h [hunk 6]
```diff
@@ -724,6 +739,59 @@ typedef struct
...
+typedef struct
+{
+    int32_t latitude;          /**< Geodetic latitude in WGS-84. Range -8388607...8388607.
+                                *   The relation between the coded number N and the latitude
+                                *   range X (in degrees) is as follows: N <= (2^24/360) * X < N + 1.
+                                *   For N = 2^23 - 1, the range is extended to include N+1.
+                                *   Range of X (in degrees) -90...90.
+                                */
+
+    int32_t longitude;         /**< Geodetic longitude in WGS-84. Range -8388607..-8388607.
```
Comment (rlubos) [inline bsdlib/include/nrf_socket.h]: `Range -8388607..-8388607.` The second value should be positive.

Comment (lemrey) [reply bsdlib/include/nrf_socket.h]: Good catch, fixed.

**Comment by lemrey** [comment]
Rebased to fix a documentation error, squashed doc commits.

**Comment by lemrey** [comment]
Matching PR to -nrf : https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1334

**Comment by rlubos** [comment]
@lemrey One more request, I missed that. Could we include the commit SHA from which the library was built in the commit message?

**Pull request merged**